libmproxy/protocol/http.py

```diff
@@ -953,6 +953,15 @@ def replace(self, pattern, repl, *args, **kwargs):
             c += self.response.replace(pattern, repl, *args, **kwargs)
         return c
 
+    def response_content(self):
+        with decoded(self.response):
+            s = self.response.content
+        return s
+
+    def response_headers(self):
+        with decoded(self.response):
+            s = str(self.response.headers) 
+        return s
```

Comment:
the headers are never encoded (only the content is, which is specified in the content-encoding header), you can just str them and remove the function here.